### PR TITLE
feat: introducing new password guideline

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- New password textfield guidelines
- Decided to do most logic in the textfield as it's guidelined

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

## Manual Testing

https://user-images.githubusercontent.com/554874/184629527-2ba00e0a-ccef-4103-b4ad-287aad82ab4e.mp4

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-337 #done

### CODE REVIEW

**Review by idoshamun — COMMENTED** [review]
Can we perhaps create a new component that will rely on the `TextField` component but just for password? By extending its functionality for example.
Given that password fields are rarely used, adding a new package to the `TextField` which is something very fundamental seems like too much to me. What do you think?

**Comment by rebelchris** [comment]
@idoshamun Refactored to a password field.
The only thing I noticed is that we currently bubble programmtic events which all trigger the validation.

See here: no user interaction was done.
(Is this something we want?)
Screenshot 2022-08-16 at 15 58 34

**Comment by idoshamun** [comment]
I think it has its reasons but I'm not sure why...
The only use case that comes to mind is when the field is already populated and we want to trigger validation in case this field is marked as required. But if initially the field is empty we don't have to bubble it

File: packages/shared/src/components/fields/PasswordField.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+export function PasswordField({
+  type,
+  ...props
+}: TextFieldProps): ReactElement {
+  const [useType, setUseType] = useState(type);
+  const [passwordStrengthLevel, setPasswordStrengthLevel] = useState<number>(0);
+  const [isValid, setIsValid] = useState<boolean>(null);
+  const hasUserAction = isValid !== null;
+  const hint = () => {
```
Comment (idoshamun) [inline packages/shared/src/components/fields/PasswordField.tsx:38]: Why do we need it as a function?

Comment (rebelchris) [reply packages/shared/src/components/fields/PasswordField.tsx]: We don't, changed it to conditions now :)

**Review by idoshamun — APPROVED** [review]

**Pull request merged**